docker/watcher.go

```diff
@@ -121,23 +124,37 @@ type Client interface {
 type WatcherConstructor func(logp *logp.Logger, host string, tls *TLSConfig, storeShortID bool) (Watcher, error)
 
 // NewWatcher returns a watcher running for the given settings
-func NewWatcher(log *logp.Logger, host string, tls *TLSConfig, storeShortID bool) (Watcher, error) {
+func NewWatcher(log *logp.Logger, host string, cfg *TLSConfig, storeShortID bool) (Watcher, error) {
 	var httpClient *http.Client
-	if tls != nil {
-		options := tlsconfig.Options{
-			CAFile:   tls.CA,
-			CertFile: tls.Certificate,
-			KeyFile:  tls.Key,
+	if cfg != nil {
+		tlsConfig := &tls.Config{
+			// Prefer TLS1.2 as the client minimum
+			MinVersion: tls.VersionTLS12,
```

Comment:
[Question]
That means TLS 1.0 and 1.1 are not accepted any more, right?

That looks like a breaking change because on our docs we state supporting TLS 1.0 and 1.1: https://www.elastic.co/guide/en/beats/filebeat/current/configuration-ssl.html#supported-protocols